catroid/src/main/java/org/catrobat/catroid/embroidery/EmbroideryList.java

```diff
@@ -32,16 +32,25 @@
 	@Override
 	public synchronized boolean add(Object o) {
 		boolean status = false;
-		PointF point = (PointF) o;
```

Comment:
if this would be:
public class EmbroideryList extends ArrayList<PointF>

the add method yould be add(PointF point) and no cast required. so that the compiler and static analysis would already complain when youd try to perform eg. add(3). while with the Object solution this would blow up at runtime.